torch/testing/_internal/common_methods_invocations.py

```diff
@@ -9544,10 +9557,12 @@ def __call__(self, opinfo, device, dtype, requires_grad, **kwargs):
     ForeachFuncInfo(
         'sin',
         foreach_inputs_sample_func(1, False, False),
+        dtypesIfCUDA=floating_and_complex_types_and(torch.half,),
```

Comment:
this means there’s more half support on CUDA vs CPU yes?

Reply:
uh, about right I'd say.
I think this is rather historical; when I started this bfloat16 wasn't as popular as today and float16 support on cpu wasn't like today.
I'll try more dtypes in foreach opinfo's (probably with some xfail/skip) after this pr